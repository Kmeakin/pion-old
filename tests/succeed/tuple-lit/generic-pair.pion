//~ run = "${PION} elab ${FILE}"

{
let fst = fun (A: Type) (B: Type) (p: {_0: A, _1: B}) => p._0;
let snd = fun (A: Type) (B: Type) (p: (A, B)) => p._1;
{}
}
