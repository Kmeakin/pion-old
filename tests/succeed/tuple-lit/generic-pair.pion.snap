exit status: 0
stdout:
{
    let fst : fun (A : Type) (B : Type) -> (A, B) -> A = fun (A : Type)
    (B : Type)
    (p : (A, B)) => p._0;
    let snd : fun (A : Type) (B : Type) -> (A, B) -> B = fun (A : Type)
    (B : Type)
    (p : (A, B)) => p._1;
    ()
} : ()

stderr:
